subprojects/testing-native/src/main/java/org/gradle/nativeplatform/test/xctest/internal/execution/XCTestExecuter.java

```diff
@@ -138,18 +147,32 @@ public void startProcessing(TestResultProcessor resultProcessor) {
 
         @Override
         public void processTestClass(TestClassRunInfo testClass) {
-            execHandle = executeTest(testClass.getTestClassName());
-            execHandle.waitForFinish();
+            StringWriter outputString = new StringWriter();
+            StringWriter errorString = new StringWriter();
+            OutputStream outputStringStream = new LineBufferingOutputStream(new WriterTextStream(outputString));
+            OutputStream errorStringStream = new LineBufferingOutputStream(new WriterTextStream(errorString));
+            execHandle = executeTest(testClass.getTestClassName(), outputStringStream, errorStringStream);
+            ExecResult result = execHandle.waitForFinish();
+            // Exit code 0 = success, Exit code 1 = failed test(s), anything else is considered an execution failure
+            if (result.getExitValue() != 0 && result.getExitValue() != 1) {
+                String failure = "Failure while running xctest executable (exit code: " + result.getExitValue() + ").";
+                LOGGER.error(failure + "\n" + displayProcessOutput(outputString.toString(), errorString.toString()));
```

Comment:
I think we should try to avoid showing the scraped output here. If the report has the useful output in it, I think it's better to direct people towards that. 

Maybe it makes sense to show stderr if we know it's going to be something really useful most of the time.

Reply:
This output makes its way into the test report now.